### PR TITLE
Route `tasks/self` for fetching all tasks of the user

### LINKED ISSUE #151: New route for fetching all the tasks of the user.
Create a new route `tasks/self` to fetch all the tasks of the user.

### PR DESCRIPTION
**Pull request by Rucha1499**
created a new route `tasks/self` for fetching all the tasks of the  user. Fixes #151

### CODE REVIEW

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]
LGTM.
@Rucha1499 , can you add tests for this?
Also need to generate the API schema for the updated routes.

File: controllers/tasksController.js [hunk 1]
```diff
@@ -38,6 +38,20 @@ const fetchTasks = async (req, res) => {
   }
 }
 
+const getSelfTasks = async (req, res) => {
```
Comment (swarajpure) [inline controllers/tasksController.js:41]: Please add JSDoc

**Review by swarajpure — CHANGES_REQUESTED** [review]

File: controllers/tasksController.js (hunk 1 above)
Comment (Rucha1499) [reply controllers/tasksController.js:41]: Okay! Thank you.

**Comment by Rucha1499** [comment]
> LGTM.
> @Rucha1499 , can you add tests for this?
> Also need to generate the API schema for the updated routes.

Okay! I will do the changes. 👍

**Review by swarajpure — DISMISSED** [review]
Looks good to me!

**Review by sumitd94 — APPROVED** [review]
Thanks, Looks good to me!

**Review by ankurnarkhede — APPROVED** [review]

File: models/tasks.js [hunk 2]
```diff
@@ -65,22 +65,49 @@ const fetchTask = async (taskId) => {
...
 
 /**
- * Fetch all tasks of a user
+ * Fetch all the active and blocked tasks of the user
  *
  * @return {Promise<tasks|Array>}
  */
 
-const fetchUserTasks = async (username) => {
+const fetchUserActiveAndBlockedTasks = async (username) => {
```
Comment (sumitd94) [inline models/tasks.js:73]: This is causing a production issue

**Pull request merged**